datadog_checks_dev/datadog_checks/dev/plugin/pytest.py

```diff
@@ -105,6 +105,18 @@ def dd_environment_runner(request):
     metadata.setdefault('env_vars', {})
     metadata['env_vars'].update(get_env_vars(raw=True))
 
+    # Inject any log configuration
+    logs_config = get_state('logs_config', [])
+    if logs_config:
+        config = format_config(config)
+        config['logs'] = logs_config
+
+    # Mount any volumes for Docker
+    if metadata['env_type'] == 'docker':
+        docker_volumes = get_state('docker_volumes', [])
+        if docker_volumes:
+            metadata.setdefault('docker_volumes', []).extend(docker_volumes)
```

Comment:
Thought: would it make sense to rely on `metadata` to retrieve the logs configurations and docker volumes?

Maybe by moving the logs mounting behavior *outside* of `docker_run()`, and into its own utility that is able to make any metadata available to the caller:

```python
with docker_run(...), mount_logs() as logs_e2e_metadata:
    # Caveat: would need some sort of "deep update" here,
    # to prevent overriding existing `logs` and `docker_volumes` entries.
    e2e_metadata.update(logs_e2e_metadata)
    yield config, e2e_metadata
```

To allow this, `mount_logs()` would yield something like `{"logs_config": [...], "docker_volumes": [...]}`, and we would access them here as `metadata["logs_config"]` and `metadata["docker_volumes"]`. (`logs_config` could be named something else, e.g. a list of `mounted_log_paths` originating from the `path` that's currently mutated in-place by `shared_logs()`.)

But this way, no need for environment variable hacks? 😄

Reply:
Resolving, I'm now fine with keeping the `save_state()`/`get_state()` mechanism which was added via a previous PR.